app/org/sagebionetworks/bridge/models/surveys/SurveyRule.java

```diff
@@ -98,21 +133,26 @@ public boolean equals(Object obj) {
                Objects.equals(operator, other.operator) &&
                Objects.equals(value, other.value) &&
                Objects.equals(endSurvey, other.endSurvey) &&
-               Objects.equals(assignDataGroup, other.assignDataGroup);
+               Objects.equals(assignDataGroup, other.assignDataGroup) &&
+               Objects.equals(dataGroups, other.dataGroups);
     }
 
     @Override
     public String toString() {
         return "SurveyRule [operator=" + operator + ", value=" + value + ", skipToTarget=" + skipToTarget
-                + ", endSurvey=" + endSurvey + ", assignDataGroup=" + assignDataGroup + "]";
+                + ", endSurvey=" + endSurvey + ", assignDataGroup=" + assignDataGroup + ", dataGroups=" + 
+                dataGroups + "]";
```

Comment:
Minor: displayIf and displayUnless? Also, what happens if any of these are null?